libraries/SITL/SIM_JEDEC.cpp

```diff
@@ -0,0 +1,192 @@
+#include "SIM_JEDEC.h"
+
+#include <errno.h>
+#include <unistd.h>
+
+#include <AP_HAL_SITL/AP_HAL_SITL.h>
+
+using namespace SITL;
+
+extern HAL_SITL& hal;
+
+void JEDEC::open_storage_fd()
+{
+    if (storage_fd != -1) {
+        AP_HAL::panic("Should not have been called");
+    }
+    
+    const char *filepath = filename();
+    uint32_t flags = O_RDWR|O_CLOEXEC;
+    storage_fd = open(filepath, flags, 0777);
+
+    if (storage_fd == -1) {
+        flags |= O_CREAT;
+        storage_fd = open(filepath, flags, 0777);
+    }
+
+    // still -1 ??
+    if (storage_fd == -1) {
+        AP_HAL::panic("open(%s): %s", filepath, strerror(errno));
+    }
+    if (ftruncate(storage_fd, storage_size()) != 0) {
+        AP_HAL::panic("truncate(%s): %s", filepath, strerror(errno));
+    }
+}
+
+void JEDEC::sector4k_erase (uint32_t addr)
+{
+    uint32_t fill_length = get_page_size()*get_page_per_sector();
```

Comment:
How big can this get?

We don't want 4k on the stack; try to limit any fill buffer to, say, 512 bytes.

Reply:
I have split sector erase into multiple page erases. Now it used a fill buffer of 256 bytes only.